controllers/dellcsivolumegroupsnapshot_controller.go

```diff
@@ -329,18 +340,80 @@ func (r *DellCsiVolumeGroupSnapshotReconciler) processResponse(ctx context.Conte
 		SnapshotGroupID: res.SnapshotGroupID,
 		Snapshots:       snaps,
 		CreationTime:    metav1.Time{Time: *createdAt},
-		ReadyToUse:      true,
-		Status:          common.EventStatusComplete,
+		ReadyToUse:      ok,
+		Status:          status,
+	}
+	// save status after processing VG response
+	if err := r.Status().Update(ctx, vg); err != nil {
+		log.Error(err, "VG Snapshotter vg created ok unable to update VG status while processing response")
+		r.EventRecorder.Eventf(vg, common.EventTypeWarning, common.EventReasonUpdated, "Failed vg create: %s error: %s", groupName, err.Error())
+		snapErrors = append(snapErrors, err)
+	}
+	if len(snapErrors) > 0 {
+		msg := " "
+		for _, e := range snapErrors {
+			msg += e.Error() + " "
+		}
+		return ctrl.Result{}, errors.New("VG Snapshotter Process Response failed " + msg)
 	}
-
 	return ctrl.Result{}, nil
 }
 
+func (r *DellCsiVolumeGroupSnapshotReconciler) checkReadyToUse(ctx context.Context, ns string, snapshots string) (bool, error) {
+	snaps := strings.Split(snapshots, ",")
+	for _, snapName := range snaps {
+		log.Info("VG check snaps", "name=", snapName)
+		//get snap
+		snap := new(s1.VolumeSnapshot)
+		nameSpacedName := t1.NamespacedName{
+			Namespace: ns,
+			Name:      snapName,
+		}
+		_ = r.Get(ctx, nameSpacedName, snap)
+		if snap.Name != "" && snap.Status != nil {
+			status := *snap.Status.ReadyToUse
+			if !status {
+				log.Info("VG check snap ReadyToUse", "ReadyToUse", "false")
+				return false, nil
+			}
+			log.Info("VG check snap ready", snapName, status)
+		} else {
+			log.Info("VG check snap", "not ready", snapName)
+			continue
+		}
+	}
+	return true, nil
+}
+
+func (r *DellCsiVolumeGroupSnapshotReconciler) checkSnapshotStatus(ctx context.Context, ns string, snaps string) (bool, error) {
+	timeout := time.After(10 * time.Second)
+	ticker := time.Tick(500 * time.Millisecond)
+	// Keep trying until we're timed out or get a result/error
+	for {
+		select {
+		// Got a timeout! fail with error
+		case <-timeout:
+			return false, errors.New("timed out")
+		// Got a tick, we should check on checkReadyToUse()
+		case <-ticker:
+			log.Info("VG check snap", "snaps", snaps)
+			ok, err := r.checkReadyToUse(ctx, ns, snaps)
+			if err != nil {
+				// We may return, or ignore the error
```

Comment:
What is the purpose of these comments, Judging by the code we are returning error correct?

Reply:
to understand how polling is done in go with ticker --this is new to team

Reply:
I am not trying to nitpick, nor trying to add more work. It seems fairly straight-forward looking at the code that there is a for loop and that exits on timeout as the exit condition, that this is a polling section. I am not in favor or comments when they dont add value, thats just my opinion. Others feel there is value in these comments, then perfectly fair. you can ignore my comments.